docs/source/data/subscriptions.mdx

```diff
@@ -354,84 +440,42 @@ Context object for the WebSocket connection. This is _not_ the `context` object
 
 ### Example: Authentication with `onConnect`
 
-On the client, `SubscriptionsClient` supports adding token information to `connectionParams` ([example](https://www.apollographql.com/docs/react/advanced/subscriptions/#authentication-over-websocket)) that will be sent with the first WebSocket message. In the server, all GraphQL subscriptions are delayed until the connection has been fully authenticated and the `onConnect` callback returns a truthy value.
+On the client, `SubscriptionsClient` supports adding token information to `connectionParams` ([example](https://www.apollographql.com/docs/react/advanced/subscriptions/#authentication-over-websocket)) that will be sent with the first WebSocket message. On the server, all GraphQL subscriptions are delayed until the connection has been fully authenticated and the `onConnect` callback returns a truthy value.
 
 The `connectionParams` argument in the `onConnect` callback contains the information passed by the client and can be used to validate user credentials.
 The GraphQL context can also be extended with the authenticated user data to enable fine grain authorization.
 
 ```js
-const { ApolloServer } = require('apollo-server');
-const { resolvers, typeDefs } = require('./schema');
-
-const validateToken = authToken => {
+async function validateToken(authToken) {
   // ... validate token and return a Promise, rejects in case of an error
 };
 
-const findUser = authToken => {
-  return tokenValidationResult => {
+function findUser(authToken) {
+  return async function (tokenValidationResult) {
     // ... finds user by auth token and return a Promise, rejects in case of an error
   };
 };
 
-const server = new ApolloServer({
-  typeDefs,
-  resolvers,
-  subscriptions: {
-    onConnect: (connectionParams, webSocket) => {
-      if (connectionParams.authToken) {
-        return validateToken(connectionParams.authToken)
-          .then(findUser(connectionParams.authToken))
-          .then(user => {
-            return {
-              currentUser: user,
-            };
-          });
+SubscriptionServer.create({
+  schema,
+  execute,
+  subscribe,
+  async onConnect(connectionParams, webSocket) {
+    if (connectionParams.authToken) {
+      try {
+        const validationResult = await validateToken(connectionParams.authToken);
+        const currentUser = await findUser(connectionParams.authToken)(validation);
```

Comment:
This isn't your fault but like... why this weird nested thing? Why not just write a `findUser` that takes two arguments?

At least you fixed the bizarro thing in the original code where the validation result was completely ignored.  But I think findUser can just take two arguments.

Though also you could just have one function here, not validateToken and findUser?  Like I have no idea why you would call two functions in a row with the same argument

Like this whole part could be

```
  try {
    const currentUser = await findUser(connectionParams.authToken);
    return { currentUser }
  }
```

Reply:
Agree, good call out!